library/Zend/Form/Element.php

```diff
@@ -83,6 +86,20 @@ public function init()
     }
 
     /**
+     * Remove the password before rendering if the form fails in order to avoid any security issue
+     *
+     * @param  FormInterface $form
+     * @return mixed
+     * @see    \Zend\Form\Element\Password::prepareElement
+     */
+    public function prepareElement(FormInterface $form)
+    {
+        if ('password' == $this->getAttribute('type')) {
+            $this->setValue('');
```

Comment:
What would be your suggestion? Changing the view helpers? Can do that, no problem, but placing password specific code in the generic `FormInput` view helper does not seem the right way to do this...

[Line 124 `Zend\Form\View\Helper\FormInput`](https://github.com/zendframework/zf2/blob/67f098af070b29d5042e89e936604df3193d2212/library/Zend/Form/View/Helper/FormInput.php#L124)
Change this:
```php
$attributes['type']  = $this->getType($element);
$attributes['value'] = $element->getValue();
```
Into this?
```php
$type                = $this->getType($element);
$attributes['type']  = $type;
$attributes['value'] = $element->getValue();
if ('password' == $type) {
    $attributes['value'] = '';
}
```